### PR TITLE
8 integration tests and n+1 fix

### PR DESCRIPTION
**Pull request by Mhamad-Raad**
# Tests
## In this branch we have done the followings:
- Added tests to the index page of users.
- Added tests to the show page of users.
- Added tests to the index page of posts.
- Added tests to the show page of posts.

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      expect(page).to have_content('Number of posts: 4')
+    end
+
+    it 'displays the last 3 posts' do
+      visit user_path(@user1)
+      expect(page).to have_content('This is my third post')
+      expect(page).to have_content('This is my second post')
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/views/users/show.html.erb_spec.rb]: -  ### For users show page you should also have the following tests:
- [ ] Asserts you can see a button that lets me view all of a user's posts
- [ ] Tests that redirects to the post's show page when you click on a user's post
- [ ] Tests that redirects to the user's post's index page when you click on a see all posts
image

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    it 'redirects to show user properly' do
+      visit user_posts_path(@user1)
+      click_link 'Post #1'
+      expect(page).to have_content('This is my first post')
+      expect(page).to_not have_content('Please approve <3')
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/views/posts/index.html.erb_spec.rb]: - ### For posts index page you should also have the following tests coverage:
- [ ] Tests that asserts you can see the user's profile picture
- [ ] Tests that asserts you can see a post's title.
- [ ] Tests that asserts you can see some of the post's body.
- [ ] Tests that asserts you can see the first comments on a post.
- [ ] Tests that asserts you can see how many comments a post has.
- [ ] Tests that asserts you can see how many likes a post has.
- [ ] Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
- [ ] Tests that redirects to the post's show page when you click on a post
image

File: spec/views/posts/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      expect(page).to have_content('First comment! Wow')
+    end
+
+    it 'should render comments corectly' do
+      visit user_post_path(@user1, @post1)
+
+      expect(page).to have_content('Comments (3):')
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/views/posts/show.html.erb_spec.rb]: - ### For posts show page  you should also have the following tests:
- [ ] Tests that asserts you can see who wrote the post.
- [ ] Tests that asserts you can see the post body.
- [ ] Tests that asserts you can see the username of each commentor.
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Mhamad-Raad                              :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with writing the majority of tests :+1: 

 ## Required Changes 
- [ ] Check the comments below :arrow_double_down: 
- _**If you are struggling with tests, use this [capybara cheat sheet](https://gist.github.com/zhengjia/428105)**_


 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Oscar, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- linters are passing

 ## Required Changes ♻️
- [ ] it looks like your forgot to push the changes after the requested changes by the Previous CR please make sure to push all the changes as requested.
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @OscarFMdev 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Mhamad-Raad** [comment]
hi there @AnselemOdims I see that you have approved the PR on GitHub but it still does not show submit on Microverse, can you please check if you approved there as well?

**Pull request merged**